### PR TITLE
Support 'op_partial_names' in config_list

### PR DESCRIPTION
**Pull request by Fiascolsy**
You now could use 'op_partial_names' instead of 'op_names' in config_list

### CODE REVIEW

File: nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py [hunk 1]
```diff
@@ -48,27 +48,31 @@
...
+    SchemaOptional('op_partial_names'): [str]
 }
 
 INTERNAL_SCHEMA = {
     'total_sparsity': And(float, lambda n: 0 <= n < 1),
     SchemaOptional('max_sparsity_per_layer'): {str: float},
     SchemaOptional('op_types'): [str],
-    SchemaOptional('op_names'): [str]
+    SchemaOptional('op_names'): [str],
+    SchemaOptional('op_partial_names'): [str]
```
Comment (J-shang) [inline nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py:75]: `op_partial_names` should not in `INTERNAL_SCHEMA`

Comment (Fiascolsy) [reply nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py]: OK, I got it. And next I will add something in need in 'config_validation.py' as well.

File: nni/algorithms/compression/v2/pytorch/utils/config_validation.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def validate_op_types(model, op_types, logger):
 
 
 def validate_op_types_op_names(data):
-    if not ('op_types' in data or 'op_names' in data):
-        raise SchemaError('Either op_types or op_names must be specified.')
+    if not ('op_types' in data or 'op_names' in data or 'op_partial_names' in data):
+            raise SchemaError('At least one of the followings must be specified: op_types, op_names or op_partial_names.')
+
+    if 'op_partial_names' in data:
+        if 'op_names' in data:
```
Comment (J-shang) [inline nni/algorithms/compression/v2/pytorch/utils/config_validation.py:63]: use `and`

File: nni/algorithms/compression/v2/pytorch/utils/pruning.py [hunk 3]
```diff
@@ -21,6 +21,21 @@ def config_list_canonical(model: Module, config_list: List[Dict]) -> List[Dict]:
...
+            for partial_name in config['op_partial_names']:
+                for name, _ in model.named_modules():
+                    if partial_name in name:
+                        op_names.append(name)
+            if 'op_names' in config:
+                config.pop('op_partial_names')
+                config['op_names'].extend(op_names)
+                config['op_names'] = list(set(config['op_names']))
+            else:
+                config['op_names'] = config.pop('op_partial_names')
```
Comment (J-shang) [inline nni/algorithms/compression/v2/pytorch/utils/pruning.py:36]: why `config['op_names'] = config.pop('op_partial_names')`? seems this `config['op_names']` is never used.

Comment (Fiascolsy) [reply nni/algorithms/compression/v2/pytorch/utils/pruning.py]: some legacy logic parts have crushed and I optimized this jut now : (

**Review by J-shang — APPROVED** [review]

**Review by acured — APPROVED** [review]

**Pull request merged**